src/components/IOUConfirmationList.js

```diff
@@ -134,46 +138,39 @@ class IOUConfirmationList extends Component {
         }];
 
         this.state = {
+            sections: this.getSections(formattedParticipants),
             participants: formattedParticipants,
+            selectedParticipants: formattedParticipants,
         };
 
+        this.allOptions = OptionsListUtils.flattenSections(this.state.sections);
```

Comment:
While I understand the reasoning here, the value returned from `OptionsListUtils.flattenSections` is not the same as the flat set of options that the sections are generated from. This is explained by the JSDoc for `flattenSections`:

```js
/**
 * Flattens an array of sections into a single array of options.
 * Each object in this array is enhanced to have:
 *
 *   1. A `sectionIndex`, which represents the index of the section it came from
 *   2. An `index`, which represents the index of the option within the section it came from.
 *
 * @param {Array<Object>} sections
 * @returns {Array<Object>}
 */
```

So I agree this is dealing with some logic that's pretty tightly coupled to the front-end / SectionList component, but I want a util for the sake of DRY. Maybe I could lean this up just by creating a separate `SectionListUtils` file?

Reply:
Got it. So the `flattenSections()` method is also "enhancing". I missed that because the name doesn't imply any changes to the items besides flattening them.

>So I agree this is dealing with some logic that's pretty tightly coupled to the front-end / SectionList component, but I want a util for the sake of DRY. Maybe I could lean this up just by creating a separate SectionListUtils file?

Agree that it better than `OptionsListUtils`, but curious to get your thoughts on this...

It looks like IOUConfirmationList and OptionsSelector have a few things in common:

- Both used `OptionsListUtils.flattenSections()`
- Both setup a similar subscriber for the "enter" key
- Both have the note about `scrollToIndex()` throwing an error and the "adjusted index" logic 

We could put all those things into utils so they can be DRY. Or take a slightly harder route of asking whether we can create a new component that splits the difference and use that in both `OptionsSelector` and `IOUConfirmationList`. That said, I haven't looked into whether it makes sense or what it would take - just a feeling I've got.

Reply:
I put a lot of work into DRYing up `IOUConfirmationList` and `OptionsSelector`, and so was able to extract `flattenSections` into `OptionsSelector` only 👍